projects/gnoland/gno.land/p/eve000/event/event.gno

```diff
@@ -120,66 +120,61 @@ func (evt *Event) AddSession(sess *component.Session) {
 	}
 	// TODO: maybe Pad3 can accept interface so extras stringconv isn't needed here
 	id := component.Pad3(strconv.Itoa(evt.storage.Sessions.Size()))
-	evt.storage.Sessions.Set(AvlKey("session", id), sess)
+	evt.storage.Sessions.Set(component.Pad3(id), sess)
 }
 
 func (evt *Event) GetSpeaker(id string) *component.Speaker {
-	s, ok := evt.storage.Speakers.Get(AvlKey("speaker", id))
+	s, ok := evt.storage.Speakers.Get(component.Pad3(id))
 	if !ok {
 		panic("speaker not found: id=" + id)
 	}
 	return s.(*component.Speaker)
 }
 
 func (evt *Event) GetLocation(id string) *component.Location {
-	l, ok := evt.storage.Locations.Get(AvlKey("location", id))
+	l, ok := evt.storage.Locations.Get(component.Pad3(id))
 	if !ok {
 		panic("location not found: id=" + id)
 	}
 	return l.(*component.Location)
 }
 
 func (evt *Event) GetSession(id string) *component.Session {
-	s, ok := evt.storage.Sessions.Get(AvlKey("session", id))
+	s, ok := evt.storage.Sessions.Get(component.Pad3(id))
 	if !ok {
 		panic("session not found: id=" + id)
 	}
 	return s.(*component.Session)
 }
 
-// REVIEW: is this used?
-func (evt *Event) GetSessions() []*component.Session {
-	if evt.storage == nil || evt.storage.Sessions == nil {
-		return nil
-	}
-	var sessions []*component.Session
-	evt.storage.Sessions.IterateByOffset(0, evt.storage.Sessions.Size(), func(_ string, val any) bool {
-		s := val.(*component.Session)
-		sessions = append(sessions, s)
-		return false
-	})
-	return sessions
-}
 func (evt *Event) Flyer() *component.Flyer {
 	flyer := &component.Flyer{
 		Name:        evt.Name,
-		Location:    evt.Location,
+		Location:    nil,
 		StartDate:   evt.StartDate,
 		EndDate:     evt.EndDate,
 		Status:      evt.Status,
 		Description: evt.Description,
-		Sessions:    evt.Sessions,
-		Images:      evt.Images,
+		Sessions:    nil,
+		Images:      append([]string(nil), evt.Images...),
+	}
+	if evt.Location != nil {
+		loc := *evt.Location
+		flyer.Location = &loc
+	}
+	if evt.Sessions != nil {
+		flyer.Sessions = make([]*component.Session, len(evt.Sessions))
+		for i, s := range evt.Sessions {
+			if s != nil {
+				sessionCopy := *s
+				flyer.Sessions[i] = &sessionCopy
+			}
+		}
```

Comment:
NOTE: we change to explicitly make a copy of pointer vars - making flyer a safe object to share with other realms